core/backend/src/CheckpointManager.ts

```diff
@@ -158,9 +160,23 @@ export class V2CheckpointManager {
 
       throw new IModelError(attachResult.result, error);
     }
+    const sasTokenExpiry = new URLSearchParams(args.auth).get("se");
+    if (sasTokenExpiry) {
+      const expiresIn = Date.parse(sasTokenExpiry) - Date.now();
+
+      this.reattachTimeouts[CheckpointManager.getKey(checkpoint)] = setTimeout(async () => {
+        await this.attach(checkpoint);

```

Comment:
This isn't safe - there's no guarantee the `AuthorizedClientRequestContext` in `checkpoint` will still be authorized when that timer fires (actually it's probably so far in the future it almost certainly won't...)

Reply:
good point. And the only way to get a correct token is from the frontend? In that case, I guess force-closing the SnapshotDb is the only solution...

Reply:
bill, isn't that why we decided that we're going to have to use client credentials to get the sas token?

Reply:
@wgoehrig what's the plan for obtaining a token to ask for the SAS key?